### PR TITLE
Add sources and fix

### LINKED ISSUE #10: MangaParser for specific theme
### Describe your suggested feature

Hii can you please make MangaParser for this theme asurascans.com just like you made for madara theme.

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/nv95/Kotatsu/issues/new/choose).
- [X] I have updated the app to version **[3.2](https://github.com/nv95/Kotatsu/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #44: Mnhaestate :add source
### Source name

Mnhaestate

### Source link

https://mnhaestate.com

### Language

Arabic

### Other details

We hope to add it because it is a beautiful and important source 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### LINKED ISSUE #54: SadScans
### Source name

SadScans

### Source link

https://sadscans.com/

### Language

Turkish

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### LINKED ISSUE #57: Manga Denizi
### Source name

Manga Denizi

### Source link

https://mangadenizi.com/

### Language

Turkish

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### LINKED ISSUE #239: VexManga (Source request)
### Source name

VexManga

### Source link

https://vexmanga.net

### Language

Arabic

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### LINKED ISSUE #240: TeamX
### Source name

TeamX

### Source link

https://teamxnovel.com

### Language

Arabic

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### LINKED ISSUE #241: EN-ARESManga (Source request)
### Source name

EN-ARESManga

### Source link

https://en-aresmanga.com/

### Language

Arabic

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### LINKED ISSUE #242: ARESManga (Source request)
### Source name

ARESManga 

### Source link

https://aresmanga.org

### Language

Arabic

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have checked that the source does not already exist on the app.
- [X] I have checked that the source does not already exist by searching the [GitHub repository](https://github.com/KotatsuApp/kotatsu-parsers) and verified it does not appear in the code base.

### PR DESCRIPTION
**Pull request by davvarrr**
New source :

freakcomic.com
goodgirls.moe
manhwa-freak.com
manhwafreak.fr
fr.readergen.fr
unitoonoficial.com
hiperdex.com
www.beyondtheataraxia.com
mangashiina.com
summertoon.com
vexmanga.net -> close #239
en-aresmanga.com -> close #241
aresmanga.org -> close #242
manhwa-raw.com
ero18x.com
manhwa-es.com
teamxnovel.com -> close #240
-> close #44 ( the site asked for a link to this url )
sadscans.com -> close #54
1stkissmanga.blog
mangadino.top
mangaowl.blog
manga-raw.info
mangaupdates.top
manhuascan.info
manhwahub.net
ghostscan.com.br
omkomik.com
omkomik.com
www.mangadenizi.net  -> close #57
www.mangageko.com
moondaisyscans.com
sarcasmscans.com
alliedfansub.online
perf-scan.fr
rackusreads.com

fix :
url -> 
Mangaschan
MasterKomik
Swa Team
cerisescan
Asura Scans en
Mgkomik
Shinigami
Nettruyenmax
Nettruyen
Random Scans
manhwalist.com
niverafansub.com -> niverafansub.co
ozulscans.com -> ozulscans.xyz

date -> Hentaizone

Legacy Scans ( the source has changed template )

remove sources :
MangaGreat -> (.com now redirects to .org)
I decided to leave the ".com" and change the domain name, as this source is the oldest on the app and therefore more likely to be used by users.

NinjaScan

close  #10 (All sites requested its either added or hs)

### CODE REVIEW

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val tag = tags.oneOrThrowIfMany()
+
+		val url = buildString {
+			append("https://$domain")
+			if (!tags.isNullOrEmpty()) {
+				append("/series?genre=")
+				append(tag?.key.orEmpty())
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt:34]: Does `urlEncoded` not required here?

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+				if (page > 1) {
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt:35]: This block is common for all 3 `if` branches, can be moved out

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val tag = tags.oneOrThrowIfMany()
+
+		val url = buildString {
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt:30]: Consider using `urlBuilder()` to build complex urls (this is not a strict rule but just a tip)

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/en/MangaGeko.kt [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+				)
+			},
+			author = doc.selectFirstOrThrow(".author").text(),
+			description = doc.selectFirstOrThrow(".description").html(),
+			chapters = doc.requireElementById("chapters").select("ul.chapter-list li")
+				.mapChapters(reversed = true) { i, li ->
+					val a = li.selectFirstOrThrow("a")
+					val url = a.attrAsRelativeUrl("href")
+					val name = li.selectFirstOrThrow(".chapter-title").text()
+					val dateText = li.select(".chapter-update").attr("datetime").substringBeforeLast(",")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/en/MangaGeko.kt:116]:
```suggestion
					val dateText = li.select(".chapter-update").attr("datetime").substringBeforeLast(',')
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LegacyScansParser.kt [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+				append("https://api.$domain/misc/home/search?title=")
+				append(query.urlEncoded())
+			}
+		} else {
+			buildString {
+				append("https://api.$domain/misc/comic/search/query?status=&order=&genreNames=")
+				if (!tags.isNullOrEmpty()) {
+					for (tag in tags) {
+						append(tag.key)
+						append(",")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/fr/LegacyScansParser.kt:52]:
```suggestion
						append(',')
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/all/Ero18x.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+internal class Ero18x(context: MangaLoaderContext) :
+	MadaraParser(context, MangaSource.ERO18X, "ero18x.com", 10) {
+	override val datePattern = "MMMM d"
+	override val sourceLocale: Locale = Locale.ENGLISH
+	override val withoutAjax = true
+
+	override suspend fun getChapters(manga: Manga, doc: Document): List<MangaChapter> {
+		val dateFormat = SimpleDateFormat(datePattern, sourceLocale)
+		return doc.body().select(selectChapter).mapChapters(reversed = true) { i, li ->
+			val a = li.selectFirst("a")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/all/Ero18x.kt:28]:
```suggestion
			val a = li.selectFirstOrThrow("a")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/all/Ero18x.kt [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+			val href = a?.attrAsRelativeUrlOrNull("href") ?: li.parseFailed("Link is missing")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/all/Ero18x.kt:29]:
```suggestion
			val href = a.attrAsRelativeUrl("href")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/es/ManhwaEs.kt [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+internal class ManhwaEs(context: MangaLoaderContext) :
+	MadaraParser(context, MangaSource.MANHWA_ES, "manhwa-es.com", 10) {
+
+	override val withoutAjax = true
+	override val datePattern = "d 'de' MMMM"
+
+	override suspend fun getChapters(manga: Manga, doc: Document): List<MangaChapter> {
+		val dateFormat = SimpleDateFormat(datePattern, sourceLocale)
+		return doc.body().select(selectChapter).mapChapters(reversed = true) { i, li ->
+			val a = li.selectFirst("a")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/es/ManhwaEs.kt:27]:
```suggestion
			val a = li.selectFirstOrThrow("a")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/es/ManhwaEs.kt [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+			val href = a?.attrAsRelativeUrlOrNull("href") ?: li.parseFailed("Link is missing")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/es/ManhwaEs.kt:28]:
```suggestion
			val href = a.attrAsRelativeUrl("href")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/id/ManhwaHub.kt [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val tag = tags.oneOrThrowIfMany()
+		val url = buildString {
+			append("https://")
+			append(domain)
+			val pages = page + 1
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/id/ManhwaHub.kt:30]: You can change the first page using `paginator.firstPage = `

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt (hunk 1 above)
Comment (davvarrr) [reply src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt:34]: With urlEncoded it doesn't work, so leave it as it is.

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt (hunk 2 above)
Comment (davvarrr) [reply src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt:35]: I thought about it too, but in any case I have to add the ? or & which is almost the same code so I think it's better to leave it like that.

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt (hunk 3 above)
Comment (davvarrr) [reply src/main/kotlin/org/koitharu/kotatsu/parsers/site/ar/TeamXNovel.kt:30]: I prefer to use buildString, which I find easier to use, but if you think it's more efficient to use urlBuilder(), don't hesitate to tell me so I can change it.

**Comment by davvarrr** [comment]
When I changed the NetTruyen URL, I saw that it was based on the wpcomics template, so I wanted to know if you wanted to add it to the template or leave it as it is, because the code was designed differently and I don't want to delete someone's work that still works.

**Pull request merged**